nussl/core/effects.py

```diff
@@ -0,0 +1,584 @@
+        shape (str): Swept wave shape, Must be "triangular" or "sinusoidal".
+        phase (float): swept wave percentage-shift for multi channel. Must be between 0 and 100.
+        interp (str): Delay Line interpolation. Must be "linear" or "quadratic".
+    Returns:
+        filter (FFmpegFilter): Resulting filter, to be called on a ffmpeg stream
+    """
+
+    _flanger_argcheck(delay, depth, regen, width, speed, phase, shape, interp)
+
+    return FFmpegFilter("flanger", delay=delay,
+                        depth=depth, regen=regen, width=width, speed=speed, phase=phase, shape=shape,
+                        interp=interp)
+
+
+def emphasis(level_in, level_out, _type="col", mode='production'):
+    """
+    https://ffmpeg.org/ffmpeg-all.html#aemphasis
+
+    Creates a FFmpegFilter object, when called on an ffmpeg stream,
+    applies a emphasis filter to the audio signal. An emphasis filter boosts frequency ranges the most 
+    suspectible to noise in a medium. When restoring sounds from such a medium, a de-emphasis filter is used
+    to de-boost boosted frequencies. 
+    Args:
+        level_in (float): Input gain
+        level_out (float): Output gain
+        _type (str): physical medium type to convert/deconvert from.
+            Must be one of the following: 
+            - "col": Columbia 
+            - "emi": EMI
+            - "bsi": BSI (78RPM)
+            - "riaa": RIAA
+            - "cd": CD (Compact Disk)
+            - "50fm": 50µs FM
+            - "75fm": 75µs FM 
+            - "50kf": 50µs FM-KF 
+            - "75kf": 75µs FM-KF 
+        mode (str): Filter mode. Must be one of the following:
+            - "reproduction": Apply de-emphasis filter
+            - "production": Apply emphasis filter
+    Returns:
+        filter (FFmpeg Filter): Resulting filter, to be called on a ffmpeg stream
+    """
+    if level_in < LEVEL_MIN or level_in > LEVEL_MAX \
+            or level_out < LEVEL_MIN or level_out > LEVEL_MAX:
+        raise ValueError(f"level_in and level_out must both be between {LEVEL_MIN} AND {LEVEL_MAX}")
+
+    allowed_types = {"col", "emi", "bsi", "riaa", "cd",
+                     "50fm", "75fm", "50kf", "75kf"}
+    if _type not in allowed_types:
+        raise ValueError(f"Given emphasis filter type is not supported by ffmpeg")
+    if mode != "production" and mode != "reproduction":
+        raise ValueError(f"mode must be production or reproduction")
+
+    # type is a reserved word in python, so kwarg dict is necessary
+    type_kwarg = {
+        'type': _type
+    }
+
+    return FFmpegFilter("aemphasis", level_in=level_in, level_out=level_out, mode=mode, **type_kwarg)
+
+
+def _compressor_argcheck(level_in, mode, reduction_ratio,
+                         attack, release, makeup, knee, link,
+                         detection, mix, threshold):
+    # The following values are taken from ffmpeg documentation
+    if (level_in < LEVEL_MIN or level_in > LEVEL_MAX or
```

Comment:
Same comment as the flanger argcheck function

Reply:
Fixed